cpp/diffraction/StaticStructureFactorDebye.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2010-2020 The Regents of the University of Michigan
+// This file is from the freud project, released under the BSD 3-Clause License.
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <stdexcept>
+
+#include "Box.h"
+#include "ManagedArray.h"
+#include "NeighborComputeFunctional.h"
+#include "NeighborQuery.h"
+#include "RDF.h"
+#include "StaticStructureFactorDebye.h"
+#include "utils.h"
+
+/*! \file StaticStructureFactorDebye.cc
+    \brief Routines for computing static structure factors.
+*/
+
+namespace freud { namespace diffraction {
+
+StaticStructureFactorDebye::StaticStructureFactorDebye(unsigned int bins, float k_max, float k_min)
+{
+    if (bins == 0)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires a nonzero number of bins.");
+    }
+    if (k_max <= 0)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires k_max to be positive.");
+    }
+    if (k_max <= k_min)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires that k_max must be greater than k_min.");
+    }
+
+    // Construct the Histogram object that will be used to track the structure factor
+    auto axes
+        = S_kHistogram::Axes {std::make_shared<util::RegularAxis>(bins, k_min, k_max)};
+    m_histogram = S_kHistogram(axes);
+    m_local_histograms = S_kHistogram::ThreadLocalHistogram(m_histogram);
+    m_min_valid_k = std::numeric_limits<float>::infinity();
+    m_structure_factor.prepare(bins);
+}
+
+void StaticStructureFactorDebye::accumulate(const freud::locality::NeighborQuery* neighbor_query,
+                                             const vec3<float>* query_points, unsigned int n_query_points, unsigned int n_total)
+{
+    auto const& box = neighbor_query->getBox();
+
+    // The r_max should be just less than half of the smallest side length of the box
+    auto const box_L = box.getL();
+    auto const min_box_length
+        = box.is2D() ? std::min(box_L.x, box_L.y) : std::min(box_L.x, std::min(box_L.y, box_L.z));
+    float const r_max = std::nextafter(float(0.5) * min_box_length, float(0));
+    const vec3<float>* points= neighbor_query->getPoints();
```

Comment:
Separate from the conversation about why we should prefer `auto` (see other thread), we should be consistent between "east const" and "west const": https://hackingcpp.com/cpp/design/east_vs_west_const.html
```suggestion
    vec3<float>* const points = neighbor_query->getPoints();
```

Reply:
Since the rest of freud uses "west const", as the article calls it, I would be in favor of updating all files in this PR to use west const instead of east to stay consistent.

Reply:
Updated to "west const".